controllers/jobs.js

```diff
@@ -4,24 +4,67 @@ const {BadRequestError, NotFoundError} = require('../errors');
 
 // Controller to handle getting all jobs
 const getAllJobs = async (req, res) => {
-    res.send('get all jobs');// response
+   const jobs = await Job.find({createdBy: req.user.userId}).sort('createdAt');
+   res.status(StatusCodes.OK).json({jobs, count: jobs.length});
 }
+
 // Controller to handle getting a single job
 const getJob = async (req, res) => {
-    res.send('get job');
+    const {
+        user:{userId}, 
+        params:{id:jobId}
+    } = req;
+
+    const job = await Job.findOne({
+        _id: jobId, 
+        createdBy: userId
+    });
+    if (!job) {
+        throw new NotFoundError(`No job with id: ${jobId}`);
+    }
+    res.status(StatusCodes.OK).json({job});
 }
 // Controller to handle creating a new job
 const createJob = async (req, res) => {
     req.body.createdBy = req.user.userId;// Add the user ID to the request body
     const job = await Job.create(req.body);// Create a new job with the request body
     res.status(StatusCodes.CREATED).json({job});// Create a new job with the request body
 }
+// Controller to handle updating a job
 const updateJob = async (req, res) => {
-    res.send('update job');
+    const {
+        body:{company, position},
+        user:{userId}, 
+        params:{id:jobId}
+    } = req;
+if (company === '' || position === '') {
```

Comment:
Nitpick:
Since empty double quotes equate to a false value, you can simplify your `if` statement:
```diff
- if (company === ""|| position === "" ){
+ if (!company || !position) {
```